src/D2L.CodeStyle.Analysis/Utils.cs

```diff
@@ -0,0 +1,26 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+
+namespace D2L.CodeStyle.Analysis {
+
+    public sealed class Utils {
+
+        private static readonly string GeneratedFolderPathSegment = Path.DirectorySeparatorChar + ".generated" + Path.DirectorySeparatorChar;

```

Comment:
This should work as long as you wouldn't get a path like `.generated\random.cs`...